suite/tests/client-interface/drreg-test.dll.c

```diff
@@ -108,6 +108,38 @@ event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst
     drreg_init_and_fill_vector(&allowed, false);
     drreg_set_vector_entry(&allowed, TEST_REG, true);
 
+#ifdef X86
+    res = drreg_init_and_fill_vector_ex(&simd_allowed, DRREG_SIMD_XMM_SPILL_CLASS, true);
+    CHECK(res == DRREG_SUCCESS, "bitvector init should work");
+    res = drreg_set_vector_entry(&simd_allowed, DR_REG_XMM0, false);
+    CHECK(res == DRREG_SUCCESS, "bitvector init should work");
+    res = drreg_set_vector_entry(&simd_allowed, DR_REG_XMM1, false);
+    CHECK(res == DRREG_SUCCESS, "bitvector init should work");
+
+    res = drreg_init_and_fill_vector_ex(&xmm0_allowed, DRREG_SIMD_XMM_SPILL_CLASS, false);
+    CHECK(res == DRREG_SUCCESS, "bitvector init should work");
+    res = drreg_set_vector_entry(&xmm0_allowed, DR_REG_XMM0, true);
+    CHECK(res == DRREG_SUCCESS, "bitvector init should work");
+
+    /* Fill with true values in order to test inversion */
+    res = drreg_init_and_fill_vector_ex(&xmm7_allowed, DRREG_SIMD_XMM_SPILL_CLASS, true);
```

Comment:
Maybe just
drreg_init_and_fill_vector_ex(&xmm7_allowed, DRREG_SIMD_XMM_SPILL_CLASS, false
[..]
res = drreg_set_vector_entry(&xmm7_allowed, DR_REG_XMM7, true);
[..]

Unless we test that the first available one is being picked, but the interface doesn't say that, just that some register of the allowed set is being picked.

Reply:
Yes, this tests that xmm7 or higher will be allocated. I wanted to avoid initing to false (like the other previous tests) in order to add variety.

Reply:
Ok. Could you add a brief comment explaining that

Reply:
Done.